source/views/menus/dev-bonapp-picker.js

```diff
@@ -47,18 +50,27 @@ export class BonAppPickerView extends React.PureComponent<Props, State> {
 		this.setState(() => ({cafeId}))
 	}
 
+	setRef = (ref: any) => (this._ref = ref)
+
 	render() {
 		return (
 			<View style={styles.container}>
 				<Toolbar onPress={() => {}}>
 					<TextInput
+						ref={this.setRef}
 						keyboardType="numeric"
-						onBlur={this.chooseCafe}
-						onChangeText={this.chooseCafe}
+						onBlur={() =>
+							this._ref ? this.chooseCafe(this._ref._lastNativeText) : null
+						}
 						style={styles.default}
-						value={this.state.cafeId}
+						value={Platform.OS === 'ios' ? this.state.cafeId : null}
+					/>
+					<Button
```

Comment:
Can't we just get rid of this button in favor of the "done" button on the keyboard?

Reply:
We're using the numeric software keyboard which does not have a "done" button. I'd advise not switching away from that keyboard because we're dealing with numeric input.

Choices:
1. Remove the "go" button and go back to live-onchange submitting
    * not a fan because I don't like hammering their API onchange
2. Keep the button as-is because it's a development utility

Reply:
I suppose a third option could be dismissing the keyboard on swipe and then firing the search. But that's more work.

Reply:
Yeah but I like the <kbd>done</kbd> button

Reply:
:p

Reply:
`returnKeyType='done'` works with "number-pad" and "numeric" on v0.47.1. Looks like we found what we wanted.